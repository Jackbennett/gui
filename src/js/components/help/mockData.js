export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
