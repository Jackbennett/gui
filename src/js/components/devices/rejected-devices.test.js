import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import Rejected from './rejected-devices';
import { undefineds } from '../../../../tests/mockData';

const mockStore = configureStore([thunk]);

describe('RejectedDevices Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      devices: {
        byStatus: {
          accepted: { total: 0 },
          rejected: { total: 0, deviceIds: [] }
        },
        filters: [],
        selectedDeviceList: [],
        limit: 500
      },
      users: {
        globalSettings: {}
      }
    });
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rejected />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
