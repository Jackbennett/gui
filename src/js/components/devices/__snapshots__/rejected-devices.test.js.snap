// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RejectedDevices Component renders correctly 1`] = `
<div
  className="tab-container flexbox column"
>
  <div
    className="loaderContainer"
  >
    <div
      className="loader"
    >
      <span
        className="dot dot_1"
      />
      <span
        className="dot dot_2"
      />
      <span
        className="dot dot_3"
      />
      <span
        className="dot dot_4"
      />
    </div>
  </div>
  <div
    className="hidden"
  >
    <p>
      There are no rejected devices
    </p>
  </div>
</div>
`;
